hal/src/nRF52840/core_hal.c

```diff
@@ -483,6 +486,14 @@ void HAL_Core_Mode_Button_Reset(uint16_t button)
 }
 
 void HAL_Core_System_Reset(void) {
+    if (nrf_wdt_started()) {
```

Comment:
@XuGuohui Do you think it makes sense to have this in the bootloader early boot sequence instead?

Reply:
The RGB will blink because of two step reset. I don't think there is other way to enter the bootloader while the watchdog is running, except for soft reset, so why not reset in system and it's easy to figure out the reset reason?

Reply:
I'm more worried about something (e.g. SoftDevice, SDK etc) calling `NVIC_SystemReset()` directly and that would bypass the whole thing altogether. You should be able to do this VERY early in the boot process of the bootloader and it should be virtually unnoticable to the user.

Reply:
And it's consistent with RTL872x-based platform, which will also reset the watchdog if soft reset is requested.

Reply:
Maybe we can keep both the check in system and bootloader.

Reply:
Might be a good idea as well, yeah. I'd strongly suggest having this check early in the bootloader just in case.